.github/workflows/Build.yml

```diff
@@ -60,117 +60,90 @@ jobs:
 
       # Checks out a copy of your repository on the ubuntu machine
       - name: Checkout code
-        uses: actions/checkout@v2
-
-      - name: Setup cache environment
-        id: extcache
-        uses: shivammathur/cache-extensions@1.6.2
-        with:
-          php-version: ${{ matrix.php-versions }}
-          extensions: ${{ env.extensions }}
-          key: ${{ env.key }}
-
-      - name: Cache PHP Extensions
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.extcache.outputs.dir }}
-          key: ${{ steps.extcache.outputs.key }}
-          restore-keys: ${{ steps.extcache.outputs.key }}
-
-      - name: Cache Composer Dependencies
-        uses: actions/cache@v2.1.6
-        with:
-          path: ~/.composer/cache/files
-          key: dependencies-composer-${{ hashFiles('composer.json') }}
-
-      - name: Setup PHP Action
-        uses: shivammathur/setup-php@2.15.0
-        with:
-          php-version: ${{ matrix.php-versions }}
-          extensions: ${{ env.extensions }}
-          coverage: xdebug
-          tools: pecl, composer
+        uses: actions/checkout@v3
 
       - name: Set Up Imagick, FFmpeg & Exiftools
         run: |
           sudo apt-get update
           sudo apt-get --fix-broken install
           sudo apt-get -y install ffmpeg libimage-exiftool-perl
 
-      - name: Get composer cache directory
-        id: composer-cache
-        run: echo "::set-output name=dir::$(composer config cache-files-dir)"
-
-      - name: Cache dependencies
-        uses: actions/cache@v2
-        with:
-          path: ${{ steps.composer-cache.outputs.dir }}
-          key: ${{ runner.os }}-composer-${{ hashFiles('**/composer.lock') }}
-          restore-keys: ${{ runner.os }}-composer-
-
       - name: setup Databases
         run: |
           sudo service mysql start
           touch database/database.sqlite
           mysql -uroot -proot -e 'create database homestead_test;'
 
+      - name: Setup PHP Action
+        uses: shivammathur/setup-php@2.15.0
+        with:
+          php-version: ${{ matrix.php-versions }}
+          extensions: ${{ env.extensions }}
+          coverage: xdebug
+          tools: pecl, composer
+
+      # Install dependencies and handle caching in one go.
+      # Dependencies need to be installed to make sure the PHPUnit classes are recognized.
+      # @link https://github.com/marketplace/actions/install-composer-dependencies
+      - name: Install Composer dependencies
+        uses: "ramsey/composer-install@v2"
+        with:
+          composer-options: "${{ matrix.mode }}"
+
       # begin of DEV
-      - name: Install Composer dependencies (dev)
-        if: ${{ matrix.mode == 'dev' }}
-        run: composer install --prefer-dist --no-interaction
+      - name: Install Composer dependencies
+        if: ${{ matrix.mode == '' }}
+        uses: "ramsey/composer-install@v2"
```

Comment:
Isn't this covered by the section above now?

Reply:
isn't this already fixed ? :'D

Reply:
https://github.com/LycheeOrg/Lychee/pull/1335/commits/56c1c8149a51ca87ca50c842df580ec7d362e1bd